cmd/repl/prompt.go

```diff
@@ -167,6 +173,32 @@ func (p *promptState) initCompleter() {
 	}
 }
 
+func (p *promptState) initHistory() {
+	p.historyFileName = filepath.Join(os.Getenv("HOME"), ".sqlflow_history")
+	f, err := os.Open(p.historyFileName)
+	if err != nil {
+		return
+	}
+	defer f.Close()
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		p.history = append(p.history, scanner.Text())
+	}
+}
+
+func (p *promptState) updateHistory() {
+	if p.statement != "" {
+		f, err := os.OpenFile(p.historyFileName, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0644)
```

Comment:
Posix seems to guarantee that writes with O_APPEND are atomic. The APUE book also describes the behavior in detail.  So we don't need a file lock here.